core/src/utils/math_util.cpp

```diff
@@ -0,0 +1,31 @@
+#include "../core/include/utils/math_util.h"
+
+/**
+* Constrain the input between a minimum and a maximum value
+* 
+* @param val  the value to be restrained
+* @param low  the minimum value that will be returned
+* @param high the maximum value that will be returned
+**/
+double clamp(double val, double low, double high){
+  if (val < low){
+    return low;
+  } else if (val > high){
+    return high;
+  }
+  return val;
+}
+/**
+* Returns the sign of a number
+* @param x
+* 
+* returns the sign +/-1 of x. 0 if x is 0
+**/
+double sign(double x){
+  if (x<0){
+    return -1;
+  } else if (x>0){
+    return 1;
+  }
+  return 0;
```

Comment:
There are some cases in Core where multiplying by 0 might cause issues (mostly with 'direction' in polar coordinates), might be best to have it return 1 by default